OpenRA.Game/Player.cs

```diff
@@ -137,8 +137,8 @@ public override string ToString()
 		public Dictionary<Player, Stance> Stances = new Dictionary<Player, Stance>();
 		public bool IsAlliedWith(Player p)
 		{
-			// Observers are considered as allies
-			return p == null || Stances[p] == Stance.Ally || p.Spectating;
+			// Human observers are considered as allies
```

Comment:
This isn't quite right.  You're attributing humanity to the wrong player, but then that's confusing with respect to the AI...

How about `// Observers are considered allies to active combatants`